python/lsst/ap/pipe/ap_pipe.py

```diff
@@ -527,27 +604,73 @@ def doDiffIm(processed_repo, sciencevisit, ccdnum, templatevisit, diffim_repo):
         config.doDecorrelation = True
 
     TODO: use coadds as templates by default, not another visit (DM-11422).
+    Some of the comments in this function are placeholders for DM-11422 work.
 
     RESULT:
-    diffim_repo/deepDiff/v+sciencevisit populated with difference images
+    diffim_repo/deepDiff/v+visit populated with difference images
     and catalogs of detected sources (diaSrc, diffexp, and metadata files)
     '''
-    if os.path.exists(os.path.join(diffim_repo, 'deepDiff', 'v' + sciencevisit)):
-        print('DiffIm has already been run for visit {0}, skipping...'.format(sciencevisit))
+    lsst.log.configure()
+    log = lsst.log.Log.getLogger('ap.pipe.doDiffIm')
+    dataId_items = re.split('[ +=]', dataId)
+    dataId_dict = dict(zip(dataId_items[::2], dataId_items[1::2]))
+    if 'visit' not in dataId_dict.keys():
+        raise RuntimeError('The dataId string is missing \'visit\'')
+    else:  # save the visit number from the dataId
+        visit = dataId_dict['visit']
+    if os.path.exists(os.path.join(diffim_repo, 'deepDiff', 'v' + visit)):
+        log.warn('DiffIm has already been run for visit {0}, skipping...'.format(visit))
         return None
     if not os.path.isdir(diffim_repo):
         os.mkdir(diffim_repo)
-    print('Running ImageDifference...')
+    log.info('Running ImageDifference...')
     config = ImageDifferenceConfig()
-    config.getTemplate.retarget(GetCalexpAsTemplateTask)
+    config.getTemplate.retarget(GetCalexpAsTemplateTask)  # visit template config
+    # config.doSelectSources = False  # coadd template config
     config.detection.thresholdValue = 5.0
     config.doDecorrelation = True
-    args = [processed_repo, '--id', 'visit=' + sciencevisit, 'ccdnum=' + ccdnum,
-            '--templateId', 'visit=' + templatevisit, '--output', diffim_repo]
+    dataId = dataId.split(' ')
```

Comment:
Why are you splitting on three characters above but only on spaces here?

Reply:
Unfortunately, `dataId` handling has to be different here because difference imaging uses `parseAndRun` while processing uses `run`. In the future, both Tasks may use `run`.